### PR TITLE
V2-3519 Block Utils Reference

### PR DESCRIPTION
**Pull request by divinentd**
https://volusion-product-team.atlassian.net/browse/V2-3519

Friendly link: https://github.com/volusion/element-documentation/blob/V2-3519-block-utils/references/block-utils/README.md

### CODE REVIEW

File: references/block-utils/README.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
+# Block Utils
+
+Blocks have access to the following utils, passed to them as props. Some are only available when the block is initially rendered by the server, and not available later on the client-side.
+
+In this document we will use `utils` to refer to `window.ElementSdk`. You may need to create a local reference to this full SDK, as often `utils` is defined as only `window.ElementSdk.client`.
```
[the remark this answers is not in the document]
Comment (divinentd) [reply references/block-utils/README.md]: Even just within the starter block I have been finding it entirely inconsistent. Sometimes it's in `props` and other times it's in `utils`, and sometimes you have a utils that's a stripped down version of the SDK with only `client`. That's why I added this warning at the top. Cameron has been discussing with my ideas for cleaning this all up, but the documents need to be the current state.

File: references/block-utils/README.md [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+| &check;            | &check;     |
+
+### `isRendering`
+
+Boolean property that is `true` when the block is rendered by the live store, or a preview of a theme, and `undefined` otherwise, for instance in Site Designer.
+
+#### `isRendering` Usage
+
+```js
+utils.pubSub.publish(utils.events.cart.openCart);
```
[the remark this answers is not in the document]
Comment (divinentd) [reply references/block-utils/README.md]: Oops!!

File: references/block-utils/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (chris-decker-volusion) [reply references/block-utils/README.md]: @divinentd This is likely an issue with the local env. I can help out with this if you want.

[the remark this answers is not in the document]
Comment (divinentd) [reply references/block-utils/README.md]: True, this is a local.js issue, but this is what comes standard in a starter block, so it's what people will be dealing with:
```const utils = {
  client: window.ElementSdk.client
}```

[the remark this answers is not in the document]
Comment (santios-volusion) [reply references/block-utils/README.md]: I think this note is confusing. I can't understand it. 

- Where do I have to create a local reference?
- What needs to be done to remove this note?

[the remark this answers is not in the document]
Comment (divinentd) [reply references/block-utils/README.md]: The current starter block setup is confusing, which is why I felt the need to include a note. The methods in this doc are sometimes accessed via `utils`, like in getDataProps.js. And in the block code you have access to `props.client`, but the rest of ElementSdk gets stripped out when it's passed to the block, so if you want anything besides client you need to reference window.ElementSdk

[the remark this answers is not in the document]
Comment (divinentd) [reply references/block-utils/README.md]: I'm working with Chris to see if we can improve the starter block so it's less confusing, and I can remove this note

[the remark this answers is not in the document]
Comment (chris-decker-volusion) [reply references/block-utils/README.md]: @santios-volusion @divinentd I've opened a PR to help resolve this in the block starter repo: https://github.com/volusion/element-BlockStarter/pull/16

[the remark this answers is not in the document]
Comment (divinentd) [reply references/block-utils/README.md]: Yup, we got it figured out. I'll get rid of this note!

**Pull request merged**